### ISSUE TITLE
Compilation fails on Ubuntu 16.04.1 x86_64

### ISSUE THREAD

**Issue by zgoda** [WeiDUorg/elkhound#4]
```
g++ -c -o agramlex.yy.o agramlex.yy.cc -g -Wall -Wno-deprecated -D__UNIX__ -O2 -DNDEBUG -D__LINUX__ -I../smbase
agramlex.yy.cc: In constructor ‘yyFlexLexer::yyFlexLexer(std::istream*, std::ostream*)’:
agramlex.yy.cc:1253:86: error: expected identifier before ‘{’ token
 yyFlexLexer::yyFlexLexer( FLEX_STD istream* arg_yyin, FLEX_STD ostream* arg_yyout ): {
                                                                                      ^
In file included from /usr/include/c++/5/iostream:40:0,
                 from agramlex.yy.cc:96:
/usr/include/c++/5/istream:606:7: error: ‘std::basic_istream<_CharT, _Traits>::basic_istream() [with _CharT = char; _Traits = std::char_traits<char>]’ is protected
       basic_istream()
       ^
agramlex.yy.cc:1253:86: error: within this context
 yyFlexLexer::yyFlexLexer( FLEX_STD istream* arg_yyin, FLEX_STD ostream* arg_yyout ): {
                                                                                      ^
In file included from /usr/include/c++/5/iostream:39:0,
                 from agramlex.yy.cc:96:
/usr/include/c++/5/ostream:384:7: error: ‘std::basic_ostream<_CharT, _Traits>::basic_ostream() [with _CharT = char; _Traits = std::char_traits<char>]’ is protected
       basic_ostream()
       ^
agramlex.yy.cc:1253:86: error: within this context
 yyFlexLexer::yyFlexLexer( FLEX_STD istream* arg_yyin, FLEX_STD ostream* arg_yyout ): {
                                                                                      ^
agramlex.yy.cc:1254:58: error: no match for call to ‘(std::ostream {aka std::basic_ostream<char>}) (std::basic_streambuf<char>*)’
  yyout(arg_yyout ? arg_yyout->rdbuf() : std::cout.rdbuf())
                                                          ^
agramlex.yy.cc:1261:39: error: expected primary-expression before ‘&’ token
 yyFlexLexer::yyFlexLexer( std::istream& arg_yyin, std::ostream& arg_yyout ):
                                       ^
agramlex.yy.cc:1261:63: error: expected primary-expression before ‘&’ token
 yyFlexLexer::yyFlexLexer( std::istream& arg_yyin, std::ostream& arg_yyout ):
                                                               ^
agramlex.yy.cc:1261:75: error: cannot call constructor ‘yyFlexLexer::yyFlexLexer’ directly [-fpermissive]
 yyFlexLexer::yyFlexLexer( std::istream& arg_yyin, std::ostream& arg_yyout ):
                                                                           ^
agramlex.yy.cc:1261:75: note: for a function-style cast, remove the redundant ‘::yyFlexLexer’
agramlex.yy.cc:1270:30: error: qualified-id in declaration before ‘(’ token
 void yyFlexLexer::ctor_common()
                              ^
agramlex.yy.cc:1298:27: error: no matching function for call to ‘yyFlexLexer::~yyFlexLexer()’
 yyFlexLexer::~yyFlexLexer()
                           ^
In file included from ../smbase/sm_flexlexer.h:28:0,
                 from agramlex.yy.cc:324:
/usr/include/FlexLexer.h:134:10: note: candidate: virtual yyFlexLexer::~yyFlexLexer()
  virtual ~yyFlexLexer();
          ^
/usr/include/FlexLexer.h:134:10: note:   candidate expects 1 argument, 0 provided
agramlex.yy.cc:1308:33: error: qualified-id in declaration before ‘(’ token
 void yyFlexLexer::switch_streams( std::istream& new_in, std::ostream& new_out )
                                 ^
agramlex.yy.cc:1320:33: error: qualified-id in declaration before ‘(’ token
 void yyFlexLexer::switch_streams( std::istream* new_in, std::ostream* new_out )
                                 ^
agramlex.lex:332:1: error: expected ‘}’ at end of input
```

GCC is 5.3.1
Bison is 3.0.4
Flex is 2.6.0

**Comment by FredrikLindgren** [comment]
Seems like ast/run_flex.pl is generating syntactically invalid output for some reason. I'll have a look.

**Comment by FredrikLindgren** [comment]
Hopefully fixed in 37b2d3682add917371dc045cf4710b6f0fca04b3
